pkg/operator/options/options_validation.go

```diff
@@ -0,0 +1,67 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"knative.dev/pkg/apis"
+)
+
+func (o Options) Validate() (errs *apis.FieldError) {
+	return errs.Also(
+		o.validateEndpoint(),
+		o.validateClusterName(),
+		o.validateVMMemoryOverheadPercent(),
+		o.validateAssumeRoleDuration(),
+	).ViaField("aws")
+}
+
+func (o Options) validateAssumeRoleDuration() (errs *apis.FieldError) {
+	if o.AssumeRoleDuration < time.Minute*15 {
+		return errs.Also(apis.ErrInvalidValue("assume-role-duration cannot be less than 15 Minutes", "assume-role-duration"))
+	}
+	return nil
+}
+
+func (o Options) validateClusterName() (errs *apis.FieldError) {
+	// TODO: Remove with karpenter-global-settings
+	if !o.setFlags["aws-cluster-name"] {
```

Comment:
```suggestion
	if !o.setFlags["cluster-name"] {
```

This is the wrong setFlag key, right?